src/simplify_dynamicquantizelinear.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/simplify_dynamicquantizelinear.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/shape.hpp>
+#include <migraphx/matcher.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/register_op.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+/*
+ *  Dynamicquantizelinear by default adds uint8_t typed zero point into a quantize linear
+ *  which needs to converted to int8 in order to avoid uint8 x int8 operations or uint8 operations
+ *  from occuring on the backend as this isn't supported by MLIR nor how we simplify our quantizable
+ *  ops.
+ */
+struct match_find_dynamicquantizelinear_convert_int8_zp
+{
+    auto matcher() const
+    {
+        return match::name("quantizelinear")(
+            match::arg(0)(skip_broadcasts(match::any())),
+            match::arg(2)(
+                skip_broadcasts(match::name("convert"),
+                                match::has_type(migraphx::shape::uint8_type).bind("convert"))));
+    }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        /* Need to modify the uint8 min/max range as well as final convert to convert to int8 */
+        auto convert_op = r.instructions["convert"];
+
+        // Ops to get q_min/q_max quickly
+        auto round_op    = convert_op->inputs().at(0);
+        auto saturate_op = round_op->inputs().at(0);
+        auto q_min       = saturate_op->inputs().at(1);
+        auto q_max       = saturate_op->inputs().at(2);
```

Comment:
I think its better to match for these instructions rather than relying on them always being is this order. As it is, this will match with any quantizelinear where the zero point is a convert instruction, and then assume these set of instructions exist in this format. Lets not assume its from the dynamicquantizelinear op and check for it in the matcher

Reply:
I'm checking based on the type being uint8_type though as part of the matcher though. Wouldn't that omit any convert->quantizelinear sequence which don't have the correct type?

Reply:
Ah fair. So I guess I'll search for the other ops in tandem then

Reply:
Added two more ops in the search for the matcher, the nearbyint and clip into the convert should be sufficient, unless you want me to check if the clip also has two inputs as well? 

I figured we know this is some kind of narrowing of data range is occuring with the additional aps and then the has_type() check should handle the type we want to look at. I don't believe the static quantization has the clip/nearbyint before a quantizelinear as well